staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -687,8 +706,14 @@ func (r *Reflector) watchList(stopCh <-chan struct{}) (watch.Interface, error) {
 			return nil, err
 		}
 		if watchListBookmarkReceived {
+			// If we go the watchlist bookmark, stop consuming events,
+			// but do not stop the watcher.
 			break
 		}
+		// It's possible that the producer closed the result channel before the
+		// watchlist bookmark was received, without first sending an error
+		// event. If so, stop the watcher.
+		w.Stop()
```

Comment:
okay, i think this is the following case https://github.com/kubernetes/kubernetes/blob/c7c422da8bb340d4fe4cc36a48a6b4751269fe91/staging/src/k8s.io/client-go/tools/cache/reflector.go#L829

Reply:
Yes. We could make that throw an error, but that seemed like a behavioral change I didn't want to introduce without knowing the consequences to callers.

Reply:
FWIW, most implementations only close the result channel after an error event or after the stop channel is closed, but there's not really any way to enforce that. Doing so in a ResultChan comment wont fix any impls that already have this bug. So I figured we need to handle it defensively here.